mobile/src/components/courses/components/category-list/category-list.tsx

```diff
@@ -12,39 +19,48 @@ type Props = {
 };
 
 const CategoryList: FC<Props> = ({ items }) => {
-  const [activeCategoryIds, setActiveCategoryIds] = useState<number[]>([]);
+  const [activeCategoryId, setActiveCategoryId] = useState<number | null>(null);
   const [filteredCategories, setFilteredCategories] = useState<
     CategoryGetAllItemResponseDto[]
   >([]);
+  const dispatch = useAppDispatch();
 
   const handlePress = (id: number): void => {
-    if (!activeCategoryIds.includes(id)) {
-      setActiveCategoryIds([...activeCategoryIds, id]);
+    if (activeCategoryId !== id) {
+      setActiveCategoryId(id);
+      dispatch(categoryActions.getCategoryById({ id }));
     } else {
-      setActiveCategoryIds(
-        activeCategoryIds.filter((activeId) => activeId !== id),
-      );
+      setActiveCategoryId(null);
+      dispatch(categoryActions.clearCategory());
     }
   };
 
   useEffect(() => {
-    if (activeCategoryIds.length) {
-      const activeSet = new Set(activeCategoryIds);
-      const activeItems = items.filter((item) => activeSet.has(item.id));
-      const notActiveItems = items.filter((item) => !activeSet.has(item.id));
-
-      setFilteredCategories([...activeItems, ...notActiveItems]);
+    if (activeCategoryId) {
+      const activeItem = items.filter((item) => item.id === activeCategoryId);
+      const notActiveItems = items.filter(
+        (item) => item.id !== activeCategoryId,
+      );
+      setFilteredCategories([...activeItem, ...notActiveItems]);
     } else {
       setFilteredCategories(items);
     }
-  }, [activeCategoryIds]);
+  }, [activeCategoryId]);
+
+  useFocusEffect(
```

Comment:
Not sure if we need this, also all focus effects should be in screens, not in children components